lib/rules/selector-max-specificity/index.mjs

```diff
@@ -80,146 +57,40 @@ const rule = (primary, secondaryOptions) => {
-
-			const nodes = pseudo.nodes[0]?.nodes ?? [];
-			const ofSelectorIndex = nodes.findIndex((child, i, children) => {
-				// Find ' of <selector>' nodes
-				return isSpace(child) && isOfSelector(children[i + 1]) && isSpace(children[i + 2]);
-			});
-
-			const ofSelector = nodes[ofSelectorIndex + 3];
-
-			if (!ofSelector || !ofSelector.value) return pseudo;
-
-			if (!isSelectorIgnored(ofSelector.value)) return pseudo;
-
-			const copy = pseudo.clone();
-			const rootSelector = copy.nodes[0];
-
-			if (rootSelector) {
-				// Remove ' of <selector>' nodes
-				rootSelector.nodes = rootSelector.nodes.slice(0, ofSelectorIndex);
-			}
-
-			return copy;
-		};
-
-		/**
-		 * Calculate the specificity of a pseudo selector including own value and children.
-		 *
-		 * @param {import('postcss-selector-parser').Pseudo} node
-		 * @returns {Specificity}
-		 */
-		const pseudoSpecificity = (node) => {
-			// `node.toString()` includes children which should be processed separately,
-			// so use `node.value` instead
-			const ownValue = node.value.toLowerCase();
-
-			if (ownValue === ':where') {
-				return zeroSpecificity();
-			}
-
-			let ownSpecificity;
-
-			if (isSelectorIgnored(ownValue)) {
-				ownSpecificity = zeroSpecificity();
-			} else if (aNPlusBOfSNotationPseudoClasses.has(ownValue.replace(/^:/, ''))) {
-				return selectorSpecificity(ignoreOfSelectorIfAny(node));
-			} else {
-				ownSpecificity = selectorSpecificity(node.clone({ nodes: [] }));
-			}
-
-			return specificitySum([ownSpecificity, maxChildSpecificity(node)]);
-		};
-
-		/**
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {boolean}
-		 */
-		const shouldSkipPseudoClassArgument = (node) => {
-			// postcss-selector-parser includes the arguments to nth-child() functions
-			// as "tags", so we need to ignore them ourselves.
-			// The fake-tag's "parent" is actually a selector node, whose parent
-			// should be the :nth-child pseudo node.
-			const parentNode = node.parent && node.parent.parent;
-
-			if (parentNode && parentNode.type === 'pseudo' && parentNode.value) {
-				const pseudoClass = parentNode.value.toLowerCase().replace(/^:/, '');
-
-				return (
-					aNPlusBNotationPseudoClasses.has(pseudoClass) || linguisticPseudoClasses.has(pseudoClass)
-				);
-			}
-
-			return false;
-		};
-
-		/**
-		 * Calculate the specificity of a node parsed by `postcss-selector-parser`.
-		 *
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {Specificity}
-		 */
-		const nodeSpecificity = (node) => {
-			if (shouldSkipPseudoClassArgument(node)) {
-				return zeroSpecificity();
-			}
-
+		/** @type {(node: import('postcss-selector-parser').Node) => Specificity | void} */
+		const isSelectorNodeIgnored = (node) => {
 			switch (node.type) {
 				case 'attribute':
 				case 'class':
 				case 'id':
 				case 'tag':
-					return simpleSpecificity(node);
-				case 'pseudo':
-					return pseudoSpecificity(node);
-				case 'selector':
```

Comment:
question

Is there a reason for this case removal?

Reply:
The old function that included a switch statement over node types also did the recursive decent down the AST. This is no longer needed.

The updated version only has a switch statement over node types to customize their specificity. Given that `selector` type nodes aren't ignorable they are omitted.

------------

Git diff seems to align these two switch statements, but they are actually unrelated in function and purpose.